### ISSUE TITLE
[Snackbar] `isAnchorViewLayoutListenerEnabled` causes infinite layout loop

### ISSUE THREAD

**Issue by lwasyl** [material-components/material-components-android#2744]
**Description:** Setting `isAnchorViewLayoutListenerEnabled` and triggering layout causes the global layout listener to update margins and request layout, triggering itself in an infinite loop

**Expected behavior:** There's no infinite `requestLayout`s happening when not needed

**Source code:** `snackbar.isAnchorViewLayoutListenerEnabled = true`

**Minimal sample app repro:** Please consider attaching a minimal sample app that reproduces the issue. This will help narrow down the conditions required for reproducing the issue, and it will speed up the bug fix process. You may attach a zip file of the sample app or link to a GitHub repo that contains the sample app.

**Android API version:** 28

**Material Library version:** 1.4.0

**Device:** emulator

To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by drchen** [comment]
Do you have a stack trace?

**Comment by drchen** [comment]
Also please check if it's still reproducible with the latest library version because we've refactored the anchor logic to a large extent.

**Comment by drchen** [comment]
I believe you can still use M2 themes/styles after updating to 1.5.x or even 1.6.x. : )

Or can you try to update for the testing purpose of this issue?

**Comment by lwasyl** [comment]
1.6.1 has the same problem, it layouts infinitely with `isAnchorViewLayoutListenerEnabled = true`

**Comment by aleksandra-krzemien** [comment]
@drchen I don't think that this change solves the issue. Or rather it seems to be introducing a new one.

When using `isAnchorViewLayoutListenerEnabled = true`, first `onGlobalLayout()` may be called before Snackbar view is added to parent. Then in `updateMargins()`, as `view.getParent() == null` we get a hint `Wait for addView() is done to update layout params`, but if I understand correctly and the comment refers to `addToTargetParent` (which calls `addView`) then it is `recalculateAndUpdateMargins()` which is called next, and as the bottomMargin didn't change, `updateMargins()` will not be called any more.

This way the margin will never get updated.

**Comment by drchen** [comment]
Hi @aleksandra-krzemien,

Thanks for the report. Can you create a new issue and link this one there? 
I'm currently pretty occupied by some other stuff but let's try to find someone who may be able to take a look.
